### PR TITLE
Fixes #771 - Resolve csrf trusted origins

### LINKED ISSUE #771: Support for CSRF_TRUSTED_ORIGINS Environment variable
### Desired Behavior

We are running netbox-docker behind an SSL termination proxy, and ran into the [issue described here](https://github.com/netbox-community/netbox/issues/8665).  It would be helpful to have this variable imported from the env/netbox.env file.

### Contrast to Current Behavior

In order to fix it, I had to directly modify configuration/configuration.yml 

### Required Changes

configuration/configuration.yml 
env/netbox.env

### Discussion: Benefits and Drawbacks

_No response_

### PR DESCRIPTION
**Pull request by SejiL**
Related Issue: #771 

## Contrast to Current Behavior

In order to fix it, I had to directly modify configuration/configuration.yml

### CODE REVIEW

File: configuration/configuration.py [hunk 1]
```diff
@@ -82,7 +84,7 @@ def _read_secret(secret_name, default = None):
 
 
 #########################
-#                       #
+#                       #http://localhost
```
Comment (cimnine) [inline configuration/configuration.py]: What's with this line?

File: env/netbox.env [hunk 2]
```diff
@@ -1,4 +1,5 @@
 CORS_ORIGIN_ALLOW_ALL=True
+CSRF_TRUSTED_ORIGINS=http://netbox.example.com/*
```
Comment (cimnine) [inline env/netbox.env]: I believe that a sensible default would just be an empty value here.

**Review by cimnine — COMMENTED** [review]
You could have also added the setting to `extra.py` in your setup. That file is meant for everything that's not in the main configuration.

Since [the upstream `configuration_example.py` file](https://github.com/netbox-community/netbox/blob/develop/netbox/netbox/configuration_example.py) does not contain that setting, I'm hesitant to have it in ours.

File: env/netbox.env (hunk 2 above)
Comment (SejiL) [reply env/netbox.env]: Removed from netbox.env

File: configuration/configuration.py (hunk 1 above)
Comment (SejiL) [reply configuration/configuration.py]: Typo fixed.

**Comment by SejiL** [comment]
I fixed your issues, Add CSRF_TRUSTED_ORIGINS in extra.py

File: configuration/extra.py [hunk 3]
```diff
@@ -53,3 +53,6 @@
 # from datetime import datetime
 # now = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
 # BANNER_TOP = f'<marquee width="200px">This instance started on {now}.</marquee>'
+
+# Allow CSRF trusted origins, e.g:
+# CSRF_TRUSTED_ORIGINS = ['http://netbox.example.com']
```
Comment (cimnine) [inline configuration/extra.py:58]: Please leave an empty line at the end of the file.

**Review by cimnine — CHANGES_REQUESTED** [review]

File: configuration/extra.py (hunk 3 above)
Comment (cimnine) [reply configuration/extra.py:58]: I'll just do it myself.

**Comment by cimnine** [comment]
Thanks. If the change only concerns `extra.py` I have no concerns merging this.

**Review by cimnine — APPROVED** [review]

**Comment by cimnine** [comment]
Manually merged in 1130ff6c6dc2043d2d54019bbd7eaa207a673bff

**Pull request closed without merging**